### PR TITLE
[FLOC-2515] Switch to requirements.txt in 1.0.x

### PR DESCRIPTION
**Pull request by tomprince**
https://clusterhq.atlassian.net/browse/FLOC-2515

These are the things that need to change (I think):
- homebrew generation
- omnibus packaging
- setup.py
- tox
- linux-install.sh
- contributing

The build results are currently meaningless because https://github.com/ClusterHQ/build.clusterhq.com/pull/111 hasn't been deployed yet. But the code should be able to be reviewed.

### CODE REVIEW

File: admin/packaging.py [hunk 1]
```diff
@@ -1181,11 +1168,16 @@ def __init__(self, distributions):
...
-    def parseArgs(self, package_uri):
+    def parseArgs(self, package_uri=None):
         """
         The Python package to install.
         """
-        self['package-uri'] = package_uri
+        if package_uri is not None:
+            # https://clusterhq.atlassian.net/browse/FLOC-2527
+            # This only exists to support the transition on buildbot.
+            # Once this is merged to all branches, this can be removed.
```
Comment (tomprince) [inline admin/packaging.py]: I'm not going to test this, since we'll throw it away soon.

File: setup.py [hunk 2]
```diff
@@ -182,37 +183,11 @@
         ],
     },
 
-    install_requires=install_requirements,    extras_require={
-        # This extra allows you to build and check the documentation for
-        # Flocker.
-        "doc": [
```
Comment (tomprince) [inline setup.py]: I don't think there was any value in separating out the individual extras.

File: setup.py [hunk 3]
```diff
@@ -37,84 +37,85 @@
 dev_requirements = [
     # flake8 is pretty critical to have around to help point out
     # obvious mistakes. It depends on PEP8, pyflakes and mccabe.
-    "pyflakes==0.8.1",
-    "pep8==1.5.7",
-    "mccabe==0.2.1",
-    "flake8==2.2.0",
+    "flake8>=2.2.0",
```
Comment (exarkun) [inline setup.py]: Why continue to specify any version here at all?

**Comment by tomprince** [comment]
Superseded by #1722.

**Pull request closed without merging**